pkg/helpers/struct_test.go

```diff
@@ -0,0 +1,124 @@
+package helpers
+
+import (
+	"encoding/base64"
+	"testing"
+	"time"
+
+	"github.com/elastic/elastic-agent-libs/mapstr"
+	"github.com/elastic/elastic-agent-shipper-client/pkg/proto/messages"
+	"github.com/stretchr/testify/require"
+)
+
+var result *messages.Value
+
+func BenchmarkCustomUnmarshal(b *testing.B) {
+	testList := mapstr.M{
+		"field1": mapstr.M{
```

Comment:
Could you encode each of the cases in the TestStructValue below into this event so we hit every type in the switch statement?